src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,462 @@
+        $curlOptions[CURLOPT_SSLKEY]       = $keyCertPath;
+        $curlOptions[CURLOPT_SSLKEYPASSWD] = $keyCertPassword;
+        $curlOptions[CURLOPT_POST]         = 1;
+
+        $config->set('curl.options', $curlOptions);
+        $this->httpClient->setConfig($config);
+
+        // don't throw exceptions for 4xx errors
+        $this->httpClient->getEventDispatcher()->addListener(
+            'request.error',
+            /** @return void */
+            function (Event $event) {
+                if ($event['response']->isClientError()) {
+                    $event->stopPropagation();
+                }
+            }
+        );
+
+        // Creating the request to be sent.
+        $httpRequest = $this->httpClient->createRequest(
+            $this->getHttpMethod(),
+            $this->getValidationUrl(),
+            $headers,
+            // ApplePayJs requires that the data be a json object.
+            json_encode($data)
+        );
+
+        /**
+         * @var static
+         */
+        return $httpRequest;
+    }
+
+    /**
+     * Overriding AbstractRequest::sendData() so that we can make a REST call instead of a SOAP call.
+     * @param array $data
+     * @psalm-suppress UndefinedMethod
+     * @return ApplePayAuthorizeResponse
+     */
+    public function sendData($data)
+    {
+        $headers = array(
+            "Content-Type" => "application/json",
+            "Accept" => "application/json"
+        );
+
+        // Create a request object to be sent.
+        $httpRequest  = $this->createClientRequest($data, $headers);
+        $httpResponse = $httpRequest->send();
+
+        // Retrieve the status code and it's corresponding reason phrase.
+        $status = array(
+            /**
+             * Have to suppress this error since Guzzle is out of scope for Psalm 
+             * 
+             * @psalm-suppress UndefinedMethod
+             */
+            'statusCode' => $httpResponse->getStatusCode(),
+            /**
+             * A human readable version of the numeric status code.
+             * Have to suppress this error since Guzzle is out of scope for Psalm .
+             * 
+             * @psalm-suppress UndefinedMethod
+             */
+            'reasonPhrase' => $httpResponse->getReasonPhrase()
+        );
+
+        // Assemble the response..
+        try {
+            $message = $httpResponse->json();
+            $response = array_merge(
+                $status,
+                ['body' => $message]
+            );
+        // If you try to parse an empty response body, error will be thrown.
+        } catch (\RunTimeException $e) {
+            $response = array_merge(
+                $status,
+                ['body' => '']
+            );
+        }
+
+        /**
+         * @var ApplePayAuthorizeResponse
+         */
+        $this->response = new static::$RESPONSE_CLASS($this, $response);
+
+        /**
+         * @var ApplePayAuthorizeResponse
```

Comment:
only one of these @var annotations is necessary since they're annotating the same variable to be the same type